modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportStartReindexJobAction.java

```diff
@@ -128,11 +115,30 @@ private void waitForReindexDone(String taskId, ActionListener<StartReindexJobAct
                 @Override
                 public void onResponse(PersistentTasksCustomMetaData.PersistentTask<ReindexJob> task) {
                     ReindexJobState state = (ReindexJobState) task.getState();
-                    if (state.getJobException() == null) {
-                        listener.onResponse(new StartReindexJobAction.Response(taskId, state.getReindexResponse()));
+                    if (state.getStatus() == ReindexJobState.Status.FAILED_TO_READ_FROM_REINDEX_INDEX) {
```

Comment:
Also for a follow-up: it could be legitimate that we cannot read from the reindex index, for instance in a full system restart situation. I wonder if we should add some kind of retries with backoff to handle this?